File: app/src/main/java/com/davismiyashiro/expenses/view/addexpense/ExpenseView.java
package com.davismiyashiro.expenses.view.addexpense;

import com.davismiyashiro.expenses.datatypes.Expense;

public interface ExpenseView {
    interface View {
        void setExpenseDescription(String desc);
        void setExpenseValue(String value);
        void setDescriptionError(boolean value);
        void setValueError(boolean value);
        void onSuccess ();
        void chooseParticipants(Expense expense);
    }

    interface UserActionsListener {
        void validateExpenseFields (String description, String value);
        void addExpense (Expense expense);
        void updateExpense (Expense expense);
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/addexpense/ExpenseView.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Davis Miyashiro
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.davismiyashiro.expenses.view.addexpense

import com.davismiyashiro.expenses.datatypes.Expense

interface ExpenseView {
    interface View {
        fun setExpenseDescription(desc: String)
        fun setExpenseValue(value: String)
        fun setDescriptionError(value: Boolean)
        fun setValueError(value: Boolean)
        fun onSuccess()
        fun chooseParticipants(expense: Expense)
    }

    interface UserActionsListener {
        fun validateExpenseFields(description: String, value: String)
        fun addExpense(expense: Expense)
        fun updateExpense(expense: Expense)
    }
}